### DISCUSSION TITLE
Cannot use :export psedo-class in scss module

### DISCUSSION THREAD

**Discussion by szaleq** [vercel/next.js#74093] [Ideas]
### Link to the code that reproduces this issue

https://github.com/szaleq/next-js-15-sass-issue-repro

### To Reproduce

1.  Create new Next.js app `yarn create next-app@latest`
2.  Install sass `yarn add sass`
3.  Add example sass code exporting variables [from documentation](https://nextjs.org/docs/app/building-your-application/styling/sass#sass-variables)

### Current vs. Expected behavior

Expected behavior: Sass variables defined using `:export {}` are correctly imported in JS using the following code:

```typescript
import vars from './vars.module.scss';
```

Current behavior: fails with the following error:

```bash
'export' is not recognized as a valid pseudo-class. Did you mean '::export' (pseudo-element) or is this a typo? at [project]/src/app/vars.module.scss.module.css:0:2
```

![image](https://github.com/user-attachments/assets/b6fe5989-a9ee-436d-b0c2-4e81deb4b9bd)

Everything works as expected with Next.js 13 and 14 (just following the steps described above but adding version to `create-next-app`, like `yarn create next-app@^13`) but fails with next@15.

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 10 Pro
  Available memory (MB): 65306
  Available CPU cores: 32
Binaries:
  Node: 18.18.2
  npm: 10.2.3
  Yarn: 4.5.3
  pnpm: 9.4.0
Relevant Packages:
  next: 15.0.3
  eslint-config-next: 15.0.3
  react: 19.0.0-rc-66855b96-20241106
  react-dom: 19.0.0-rc-66855b96-20241106
  typescript: 5.7.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Not sure, Module Resolution

### Which stage(s) are affected? (Select all that apply)

next dev (local)

**Comment by victorvianaom** [comment]
Same for me, with 2 different libs:

[excalidraw](https://github.com/excalidraw/excalidraw)
![image](https://github.com/user-attachments/assets/f5874c2f-3f93-4327-bfde-a509a78509b6)

[odometer](https://github.com/bezenson/react-odometerjs)
![image](https://github.com/user-attachments/assets/905ac227-b87d-4a99-85c8-16590c54a7b9)

**Comment by victorvianaom** [comment]
It's not a bug, turbopack does not support `:export` ICSS rule.

https://nextjs.org/docs/architecture/turbopack#unsupported-features

![image](https://github.com/user-attachments/assets/4a895c93-77bf-4181-8e02-38e3ac95e472)


---

`*` is also considered by them [invalid css](https://github.com/vercel/next.js/issues/66394#issuecomment-2149967630)

**Comment by jaandrews** [comment]
Ran into this issue myself after upgrading to nextjs 16.0.1, which defaults to using turbopack. I understand the reason it no longer works, but is there a new recommended approach to pull scss variables into the javascript, or is it just no longer possible? Current [sass documentation](https://nextjs.org/docs/app/guides/sass) only shows the method that no longer works.

**Reply by sKopheK** [reply]
wanted to add the link myself. it's contradictory what documentation says and what next build process actually does
